### PR TITLE
Factory reset: shutdown extensions

### LINKED ISSUE #7587: Factory reset: Stop all extension processes
### Problem Description

Currently, extensions may run processes (and they might run in the background). This makes it awkward for factory reset, because we should not be running processes from extensions after Rancher Desktop goes away; additionally, on Windows it is not possible to delete executables for running processes.

### Proposed Solution

We should amend factory reset to also stop all processes running out of the extension directories.

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by mook-as**
This makes us force terminate processes from extensions on factory reset, where that is defined as processes whose main executable is in the extension directory.

This also makes factory reset use some of the same code (removing shelling out to `pkill` for darwin and Linux).

Fixes #7587

### CODE REVIEW

File: src/go/rdctl/pkg/factoryreset/factory_reset_windows.go [hunk 1]
```diff
@@ -75,106 +68,9 @@ func KillRancherDesktop() error {
 		return fmt.Errorf("could not find application directory: %w", err)
 	}
 
-	var processes []uint32
-	err = directories.InvokeWin32WithBuffer(func(size int) error {
-		processes = make([]uint32, size)
-		var bytesReturned uint32
-		// We can't use `windows.EnumProcesses`, because it passes in an incorrect
```
Comment (mook-as) [inline src/go/rdctl/pkg/factoryreset/factory_reset_windows.go:82]: This was fixed in `golang.org/x/sys/windows@0.9.0`.

File: src/go/rdctl/pkg/directories/directories.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+	// Given the path to the exe, find its directory, and drop the
+	// "resources\win32\bin" suffix (possibly with another "resources" in front).
+	// On mac, we need to drop "Contents/Resources/resources/darwin/bin".
+	resultDir := filepath.Dir(exePath)
+	for _, part := range []string{"bin", platform, "resources", "Resources", "Contents"} {
+		for filepath.Base(resultDir) == part {
+			resultDir = filepath.Dir(resultDir)
+		}
+	}
+	return resultDir, nil
```
Comment (jandubois) [inline src/go/rdctl/pkg/directories/directories.go:52]: This feels a little "loose"; if I call `~/bin/rdctl` it would consider `$HOME` to be the application directory.

It probably doesn't matter, as nobody should be copying `rdctl` out of the app dir, but still feels wrong on principle.

I realize that this is just modified existing code, so the behaviour is unchanged.

File: src/go/rdctl/pkg/factoryreset/delete_data_linux.go [hunk 3]
```diff
@@ -1,15 +1,17 @@
...
 	"path/filepath"
 
 	"github.com/rancher-sandbox/rancher-desktop/src/go/rdctl/pkg/autostart"
 	"github.com/rancher-sandbox/rancher-desktop/src/go/rdctl/pkg/paths"
+	"github.com/rancher-sandbox/rancher-desktop/src/go/rdctl/pkg/process"
 	"github.com/sirupsen/logrus"
 )
 
-func DeleteData(appPaths paths.Paths, removeKubernetesCache bool) error {
+func DeleteData(ctx context.Context, appPaths paths.Paths, removeKubernetesCache bool) error {
```
Comment (jandubois) [inline src/go/rdctl/pkg/factoryreset/delete_data_linux.go:14]: Why did you add the `ctx` argument? It seems to be unused on all 3 platforms.

Comment (mook-as) [reply src/go/rdctl/pkg/factoryreset/delete_data_linux.go:14]: Mainly because having an unused context is easier to deal with than _not_ having it later when you need it. I guess I can remove it again now that it ends up unused…

**Review by jandubois — APPROVED** [review]
Thanks, LGTM

Doesn't handle subprocesses whose binaries are not stored inside the extension directory. https://github.com/rancher-sandbox/rancher-desktop-rdx-open-webui/issues/22 has more details on how this can be improved.
